### DISCUSSION TITLE
form field type 'enum' url does not change runtime

### DISCUSSION THREAD

**Discussion by lech-summary** [vitmalina/w2ui#2516] [Q&A]
good day
I have a form (w2form). Have two 'enum' type fields. 
Field1 url is defined during creation of form. Field2 url is set same as Field1 but needs to be changed dependent on Field1 value.
I have
 onChange: function(event){
  event.onComplete=function(){
  this.set("Field2", {options: {url: 'server/'+this.getValue('Field1')[0].id}});
this.refresh();
}
This changes the Field2.options.url to desired one. But http request is still executed based on initially set url (not modified).

Same setup works on 1.5. On 2.0 don't. 
Please help with solution.

w2ui -rules :).
Lech

**Comment by WillTheFarmer** [comment]
Since 2,0 - combo, list & enum controls are different and better. Once the control (object) has been instantiated changes must be made using the control's w2field object.

For your example above: Field2.w2field.options.url = 'server/'+this.getValue('Field1')[0].id}});

I use this functionality to query the w2field.options.items property when other dropdown controls change. For example:

this.get('moldid').w2field.options.items = app.moldid.filter(molds => molds.toolingid === toolingid);

Sorry for late reply but I just saw your post. Hope this helps.

**Reply by lech-summary** [reply]
Hi Will thank you for your answer. 
I applied your solution.
        
onChange: function(event){
event.onComplete=function(){
this.get(“whitness”).options.url="people/"+this.getValue("author")[0].id;
}}
Still don’t work. ☹

My case: form with two fields: author and whitness. I want list of “whitness” to exclude person selected as “author”.
Let say server with url: “people/5” returns list of people except person id: 5.
With above code I get url for whitness field properly but it still creates http request according to initial url. 
Tried generateHTML(), tried refresh(). No effect.

**Reply by WillTheFarmer** [reply]
replace:
this.get(“whitness”).options.url="people/"+this.getValue("author")[0].id;
with:
this.get(“whitness”).w2field.options.url="people/"+this.getValue("author")[0].id;

**Comment by lech-summary** [comment]
Oh...
Perfect. Works 100%.
And It was clear in your first comment. How could I miss it ???
Thank you very much. 
All the best. 
Lech.
